src/processor/processor.cpp

```diff
@@ -30,69 +26,43 @@ std::shared_ptr<FactorizedTable> QueryProcessor::execute(
     // prevOperator in the same pipeline, and decompose build and its prevOperator into another
     // one.
     auto task = std::make_shared<ProcessorTask>(resultCollector, context);
-    decomposePlanIntoTasks(lastOperator, nullptr, task.get(), context);
+    decomposePlanIntoTask(lastOperator->getChild(0), lastOperator, task.get(), context);
+    initTask(task.get());
     taskScheduler->scheduleTaskAndWaitOrError(task, context);
     return resultCollector->getResultFactorizedTable();
 }
 
-void QueryProcessor::decomposePlanIntoTasks(
+void QueryProcessor::decomposePlanIntoTask(
     PhysicalOperator* op, PhysicalOperator* parent, Task* parentTask, ExecutionContext* context) {
-    if (op->isSink() && parent != nullptr) {
+    if (op->isSink()) {
+        assert(parent != nullptr);
         auto childTask = std::make_unique<ProcessorTask>(reinterpret_cast<Sink*>(op), context);
```

Comment:
Not related to this PR, but can we rename `childTask` to `task`? It's confusing to have both `parentTask` and `childTask`, as `childTask` may refer to current task's child, or parentTask's child